### DISCUSSION TITLE
Using Expo with TVOS

### DISCUSSION THREAD

**Discussion by thevinter** [expo/expo#16149] [Archived: Questions] [answered]
Hello, I'm new to React Native so sorry if this seems like a trivial question. I currently have to port an existing React codebase to React Native and Expo seemed perfect for the job. Sadly our app will have to be distributed on iOStv, hence we need to use [react-native-tvos](https://github.com/react-native-tvos/react-native-tvos) instead of the standard `react-native`. 

Is it possible to do it with Expo as well? How would I do it? 

Thanks

EDIT: 
Just to elaborate, I followed tvos' docs and added the following to my file.
```
"react-native": "npm:react-native-tvos@latest",
```
It seems to be working but I would love some confirmation that it won't break something in the future with Expo, ty

**Comment by abdullahceylan** [comment]
To my experience, most of the features are working but one of the most important things don't work: `onFocus`, `onBlur` events for the Touchable* components. I thought I was doing something wrong with my code and spent the whole week because it was somehow working with `Expo Go` on the emulator but generated APK on a real device. I have tried running `expo run:android` rather than `expo start` to see if it's going to work by extracting the `android` folder and using the real build. No luck. This time it didn't work on the emulator either. Then I decided to use the same code with a fresh `react-native-tvos` template.

Guess what? It worked **without** `expo`.

**Comment by douglowder** [comment]
I have done some research into this. With some patches, I was able to modify the bare minimum expo template to work on tvOS. This work is still at a very early stage 😀

**Reply by nunsie** [reply]
thats such great news! following

**Comment by abdullahceylan** [comment, marked as answer]
Hey,
Sorry, I've been busy and couldn't update this discussion.
I have already made a template for `tvOS` and made it work on the simulator and real device. 

**Template:**
https://github.com/abdullahceylan/expo-template-bare-minimum-tvos

It's also published as an `npm` package.

Also, there is a discussion in https://github.com/expo/expo/pull/18970

So I'm planning to create a `config plugin` for this according to Evan's suggestion. But until that time, you can use the template above for your apps ;)

My question would be related to the `react-native-tvos` release for you @douglowder :)

What's the plan for bumping a new version according to `react-native@0.69.x`? It's `0.70.0` now though.

**Reply by douglowder** [reply]
Thank you for doing this work @abdullahceylan ! I would be interested in trying out your template, and also seeing if we can get it to work for Android TV.

**Reply by douglowder** [reply]
I would also be interested to know if you got packages like `expo-updates` to work.

**Reply by abdullahceylan** [reply]
@douglowder It's working on both the simulator and real devices. I have confirmed that it works on an Android TV as well. The most of `expo-*` packages are working but I haven't tried with `expo-updates` yet but as far as it's been used through the `prebuildCommand` option in `eas.json` file, I don't see any reason not to work.

**Reply by douglowder** [reply]
Also @abdullahceylan after a couple of bug fixes go in, I will be starting work on the 0.70.0 migration.

**Comment by grigdodon** [comment]
hey @abdullahceylan ! can your template be used for both tvOS and iOS? meaning can you develop into one expo repository for both tvOS and iOS?

also, when I am trying to install the template, I get this error when it tries to install the pods
```
Unable to find a target named `com.f6cf2474352e37afc14d7b1a7d6f0c34` in project `f6cf2474352e37afc14d7b1a7d6f0c34.xcodeproj`, did find `f6cf2474352e37afc14d7b1a7d6f0c34`.
pod exited with non-zero code: 1
```
any advice?
thanks!

**Reply by abdullahceylan** [reply]
Hi @grigdodon 

Sorry for the late reply as I've been busy with some other projects.

Unfortunately, it currently doesn't work with `tvOS` as none of the expo packages supports `tvOS`. I have updated the template -_not pushed the changes yet_- and used a hacky way to make it work but eventually, it caused some other issues.

I'll keep you posted if there would be any progress but for now, it works only on Android TV.

**Reply by JS-GitRepo** [reply]
If it only works on Android TV, is there any reason to use react-native-tvos? Sorry if this is a surface level question; fairly new to using React Native with TV.

**Reply by KaneDenzil** [reply]
Hey @abdullahceylan can you please let us know if you have any updates on if this is working for `tvOS`?
